### DISCUSSION TITLE
How to improve loading time?

### DISCUSSION THREAD

**Discussion by franpersanchez** [mmistakes/minimal-mistakes#4743] [Q&A]
I just created a personal blog using minimal-mistakes remote theme (in github pages). I was following this blog also for some guidance
https://www.janmeppe.com/

My site has a registered domain with a cloudflare CDN pointing to github server and everything is (should) correct but I noticed a key difference in the provided blog and my site (and the official site for minimal mistakes):

when accesing an already visited post, the page takes to load as if it was not previously visited (but i can see memory cache is being used) while in jan meppe blog, the loading time is almost instant. I see this improved behavior in other blogs using jekyll (but no minimal mistakes in these cases) such as:

-  http://sassdoc.com/

-  https://kittygiraudel.com/blog/

Its a little difference but a difference after all. Any ideas or guidance in how to improve loading times?

**Comment by franpersanchez** [comment]
So I found what makes Jan Meppe's blog faster than my minimal-mistakes fork...and it is really weird.
Jan has a script in the _includes/scripts.html to fetch asyncronously the mathjax library:

```html
<script type="text/javascript" async
	src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-MML-AM_CHTML">
</script>

<script type="text/x-mathjax-config">
   MathJax.Hub.Config({
     extensions: ["tex2jax.js"],
     jax: ["input/TeX", "output/HTML-CSS"],
     tex2jax: {
       inlineMath: [ ['$','$'], ["\\(","\\)"] ],
       displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
       processEscapes: true
     },
     "HTML-CSS": { availableFonts: ["TeX"] }
   });
</script>
``` 

I only have one example post in my page and no latex on it, but if I add that scrip, my web page runs like a bullet. I tried this also forking Jan Meppe's site and deleting the scripts and his blog becomes slow.

Any clue about this? How can MathJax library's script affect the loading time of my minimal mistakes blog when I dont even need that library?

**Reply by svaderia** [reply]
What I see right now is that the `script.html` Jan Mappe's blog contains is an older version. Here is the latest [scripts.html](https://github.com/mmistakes/minimal-mistakes/blob/master/_includes/scripts.html). 

The answer to the slowness(?) lies somewhere in these changes. Maybe the new changes made it somehow worse. I am no expert at this.
